source/server/BUILD

```diff
@@ -330,6 +327,9 @@ envoy_cc_library(
         "//source/extensions/filters/listener:well_known_names",
         "//source/extensions/filters/network/http_connection_manager:config",
         "//source/extensions/transport_sockets:well_known_names",
+        "//source/extensions/upstreams/http/generic:config",
+        "//source/extensions/upstreams/http/http:config",
+        "//source/extensions/upstreams/http/tcp:config",
```

Comment:
Does this mean that the extensions are always linked? Should we remove then from the bzl file?

/wait-any

Reply:
I believe they're always linked (unless some bazel magic exempts them) but the bzl file is used in a bunch of the automatic tooling so seems to be required for presubmits.

Reply:
OK maybe put a comment in the bzl file mentioning that these 3 are always linked no matter what?

Reply:
If we link this way then it shouldn't be in build_config.bzl. For TLS we do mark them as required in `https://github.com/envoyproxy/envoy/blob/master/source/extensions/all_extensions.bzl`, even they aren't in build_config.bzl it will be still linked, but can be swapped by the build_config.bzl.

Reply:
happy to remove it in theory, but in practice if I do the build fails.

As documented in https://github.com/envoyproxy/envoy/pull/11540/files protodoc.py chokes if they're not present.

@htuch is there some place for extensions which are linked in to be exempted or hard-coded so protodoc.py doesn't fail?

Reply:
Also given they're already both linked and in bzl, would you be OK with me trying to sort out tooling in a follow-up?

Reply:
Yeah I'm fine with a TODO, just mark they are not override-able/swappable but force linked

Reply:
+1 SGTM

Reply:
Not currently, protodoc is expecting to work via the extensions.bzl.